packages/synapse-react-client/src/components/ChallengeDetailPage/ChallengeDetailPage.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react'
+import ChallengeRegisterButton from '../ChallengeRegisterButton'
+import ChallengeTeamWizard from '../ChallengeTeamWizard'
+import AccessRequirementList from '../AccessRequirementList/AccessRequirementList'
+import { useGetEntityChallenge } from '../../synapse-queries'
+import { Challenge } from '@sage-bionetworks/synapse-types'
+import ConfirmationDialog from '../ConfirmationDialog'
+import { ANONYMOUS_PRINCIPAL_ID } from '../../utils/SynapseConstants'
+import { SynapseQueries } from '../..'
+
+export type ChallengeDetailPageProps = {
+  projectId: string
+}
+
+const ChallengeDetailPage = ({ projectId }: ChallengeDetailPageProps) => {
+  const [showWizard, setShowWizard] = useState<boolean>(false)
+  const [showRequirements, setShowRequirements] = useState<boolean>(false)
+  const [challenge, setChallenge] = useState<Challenge>()
+  const [showLeaveConfirm, setShowLeaveConfirm] = useState<boolean>(false)
+  const [showRegister] = useState<boolean>(true)
+  const { data: userProfile } = SynapseQueries.useGetCurrentUserProfile()
+
+  const toggleShowWizard = (b: boolean) => {
+    setShowWizard(b)
+  }
+  const onJoinClick = () => {
+    setShowRequirements(true)
+  }
+  const onLeaveClick = () => {
+    setShowLeaveConfirm(true)
+  }
+
+  const doLeaveChallenge = () => {
+    if (
+      challenge &&
+      userProfile &&
+      userProfile?.ownerId !== ANONYMOUS_PRINCIPAL_ID.toString()
+    ) {
+      /**
+       * What does it mean to leave a challenge?
+       * Simply leaving the participant team may not be enough...
+       * the user may also belong to one or more challenge team(s).
+       * Should they be removed from those?
+       * What if the team belongs to more than one challenge?
+       * Should the team be removed from the challenge instead?
+       * Only a team admin can request this however.
+       */
+      /*
+      deleteMemberFromTeam(
+        challenge.participantTeamId,
+        userProfile.ownerId,
+        accessToken,
+      )
+      */
```

Comment:
Removing a member from the challenge participant team does not remove them from any submission teams they may belong to, and leaves the user in an ambiguous state vis-a-vis the challenge. Further discussion may be required to determine what steps are required here, or whether leaving should be offered as an option at all.